AE-1874 Catch expired certificate in PDF signing

muep opened the pull request.

muep committed:
AE-1874 Catch expired certificate in PDF signing

muep committed:
AE-1874 Pass signing time into PDF service

This enhances testability of the interface

Tests needed some refactoring to keep things working as well

Juholei commented on etp-backend/src/main/clj/solita/common/certificates.clj:
```diff
@@ -22,3 +22,7 @@
             (-> certificate buddy/subject (str/split #",")))))
+
+(defn not-after [certificate]
+  (when (instance? org.bouncycastle.cert.X509CertificateHolder certificate)
```
Määrittele luokkien importit eikä koko paketin nimeä mukaan tähän

muep replied:
Näin tehtiin

Juholei commented on etp-backend/src/main/clj/solita/etp/service/energiatodistus_pdf.clj:
```diff
@@ -774,6 +774,21 @@
 
+(defn validate-not-after! [now certificate]
+  (let [not-after (-> certificate certificates/not-after)]
+    (when (.before not-after now)
```
Jos tästä tulee reflection warning, niin voisi laittaa tähän typehintin mukaan tähän jotta ei tule

muep replied:
Tuli ja laitoin vinkin

Juholei commented on etp-backend/src/test/clj/solita/etp/service/energiatodistus_pdf_test.clj:
```diff
@@ -130,9 +129,35 @@
+      (t/is (instance? clojure.lang.ExceptionInfo ex))
+      (t/is (= :expired-signing-certificate type))))
+
+  (t/is (nil? (service/validate-certificate! "Specimen-POtex"
```
Lisää (testing) tähän ympärille ja dokumentoi siihen mitä tämä testaa.

muep replied:
Tehty

muep committed:
AE-1874 Review feedback

- Import instead of referring to classes on the fly
- Add type hint to avoid reflection
- Improve readability in validate-certificate!-test

muep committed:
AE-1874 One more reflection warning
